src/components/CreateCristinUserDialog.tsx

```diff
@@ -0,0 +1,119 @@
+import { LoadingButton } from '@mui/lab';
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  TextField,
+  Typography,
+  FormControlLabel,
+  Checkbox,
+  Button,
+} from '@mui/material';
+import { ErrorMessage, Field, FieldProps, Form, Formik } from 'formik';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { CristinApiPath } from '../api/apiPaths';
+import { authenticatedApiRequest } from '../api/apiRequest';
+import { getCurrentUserAttributes } from '../api/userApi';
+import { emptyUser } from '../pages/basic_data/institution_admin/AddEmployeePage';
+import { setUser } from '../redux/actions/userActions';
+import { setNotification } from '../redux/notificationSlice';
+import { CreateCristinUser, CristinUser, FlatCristinUser, User } from '../types/user.types';
+import { isErrorStatus, isSuccessStatus } from '../utils/constants';
+import { useAuthentication } from '../utils/hooks/useAuthentication';
+import { convertToCristinUser } from '../utils/user-helpers';
+import { userValidationSchema } from '../utils/validation/basic_data/addEmployeeValidation';
+
+interface CreateCristinUserDialogProps {
+  user: User;
+}
+
+export const CreateCristinUserDialog = ({ user }: CreateCristinUserDialogProps) => {
+  const { t } = useTranslation('common');
+  const dispatch = useDispatch();
+  const { handleLogout } = useAuthentication();
+  const [acceptTermsValue, setAcceptTermsValue] = useState(false);
+
+  const createUser = async (values: FlatCristinUser) => {
+    const cristinUser: CreateCristinUser = convertToCristinUser(values);
+    const createPersonResponse = await authenticatedApiRequest<CristinUser>({
+      url: CristinApiPath.Person,
+      method: 'POST',
+      data: cristinUser,
+    });
+    if (isErrorStatus(createPersonResponse.status)) {
+      dispatch(setNotification({ message: t('feedback:error.create_user'), variant: 'error' }));
+    } else if (isSuccessStatus(createPersonResponse.status)) {
+      dispatch(setNotification({ message: t('feedback:success.create_user'), variant: 'success' }));
+      const newUserInfo = await getCurrentUserAttributes();
```

Comment:
Usikker på om dette er nok, eller om noe mer må gjøres for å synce navn fra cristin...

Reply:
Litt mer kompleks denne: https://unit.atlassian.net/browse/NP-9082?focusedCommentId=79837

Bør helst logge inn på nytt. Prøver først å sjekke om det holder å sette returnert id (og navn?) i redux

Reply:
virker fint om vi bruker responsen videre, så lar det være slik.